subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepository.java

```diff
@@ -101,13 +101,18 @@ public IncrementalTaskInputs getInputChanges() {
 
             IncrementalTaskInputs taskInputs;
             if (outputsRemoved || getStates().isRebuildRequired()) {
-                taskInputs = instantiator.newInstance(RebuildIncrementalTaskInputs.class, task, history.getCurrentExecution());
+                taskInputs = instantiator.newInstance(RebuildIncrementalTaskInputs.class, task, getInputFileFingerprints());
             } else {
                 taskInputs = instantiator.newInstance(ChangesOnlyIncrementalTaskInputs.class, getStates().getInputFilesChanges());
             }
             return taskInputs;
         }
 
+        @Override
+        public Iterable<? extends FileCollectionFingerprint> getInputFileFingerprints() {
```

Comment:
Should this be called `getCurrentInputFileFingerprints()`?

Reply:
Renamed.